app/models/assessment.rb

```diff
@@ -53,7 +53,9 @@ def can_decline?
     sections.any? { |section| section.state == :action_required }
   end
 
-  alias_method :can_request_further_information?, :can_decline?
+  def can_request_further_information?
+    can_decline? && sections.none?(&:decline?)
```

Comment:
This seems a bit confusing with all the `decline` mentions.

Reply:
Good point, I've separated it in to two lines, does that help?

Reply:
Yeah. I think that's better. What about splitting a couple of private methods out? Something like:

```
  def can_decline?
    action_required?
  end

  def can_request_further_information?
    action_required? && !must_decline?
  end

  def must_decline?
    sections.any?(&:declines_assessment)
  end

  def action_required?
    sections.any? { |section| section.state == :action_required }
  end

```